XMonad/Prompt.hs

```diff
@@ -619,7 +619,8 @@ eventLoop handle stopAction = do
         []  -> do
                 d <- gets dpy
                 io $ allocaXEvent $ \e -> do
-                    maskEvent d (exposureMask .|. keyPressMask) e
+                    nextEvent d e
```

Comment:
I don't think this is safe as it will discard all events, including map notifications, configure requests, etc.

Reply:
Ah you're right, of course.  I suppose, since we're having troubles with presses that trigger a grab, we only need to add a `buttonPressMask` to the `maskEvent` that was there before and it should work.

I just did exactly that and it does indeed still work (in xephyr, at least)